fxa-rust-client/ffi/src/lib.rs

```diff
@@ -90,3 +106,71 @@ pub extern "C" fn fxa_get_sync_keys(fxa: *mut FirefoxAccount) -> *mut SyncKeysC
     };
     Box::into_raw(Box::new(sync_keys.into()))
 }
+
+/// The caller should de-allocate the result using fxa_oauth_flow_free after use.
+#[no_mangle]
+pub extern "C" fn fxa_begin_oauth_flow(
+    fxa: *mut FirefoxAccount,
+    redirect_uri: *const c_char,
+    scopes: *const c_char,
+    wants_keys: bool,
+) -> *mut OAuthFlowInitC {
+    let fxa = unsafe {
+        assert!(!fxa.is_null());
+        &mut *fxa
+    };
+    let redirect_uri = c_char_to_string(redirect_uri);
+    let scopes = c_char_to_string(scopes);
+    let scope: Vec<&str> = scopes.split(" ").collect();
```

Comment:
Ditto here re: naming consistency, where we have "scopes" naming a single string and "scope" naming a list of strings.